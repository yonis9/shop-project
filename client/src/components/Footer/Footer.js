import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect'

import { currentUserSelector } from '../../redux/user/user-selectors'
import { signOutStart } from '../../redux/user/user-actions'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faHeart } from '@fortawesome/free-solid-svg-icons'
import { faLinkedin, faGithubSquare } from '@fortawesome/free-brands-svg-icons'

import { withRouter } from 'react-router-dom'

import {
    FooterContainer,
    NavigationContainer,
    LeftNavigation,
    RightNavigation,
    Navlink,
    FooterEnd,
    P,
    IconContainer,
    LinkContainer
} from './Footer.styles';

export const Footer = ({history, currentUser, signOutStart}) => (
    <FooterContainer>
        <NavigationContainer>
            <LeftNavigation>
                <Navlink onClick={() => history.push('/shop/mens')}>Mens</Navlink>
                <Navlink onClick={() => history.push('/shop/womens')}>Womens</Navlink>
                <Navlink onClick={() => history.push('/shop/hats')}>Hats</Navlink>
                <Navlink onClick={() => history.push('/shop/jackets')}>Jackets</Navlink>
                <Navlink onClick={() => history.push('/shop/sneakers')}>Sneakers</Navlink>
            </LeftNavigation>
            <RightNavigation>
                {
                    currentUser ? 
                    <Navlink onClick={signOutStart}>Sign Out</Navlink> :
                    <Navlink onClick={() => history.push('/signin')}>Sign In</Navlink>
                }
                <Navlink onClick={() => history.push('/checkout')}>Checkout</Navlink>
                
            </RightNavigation>
        </NavigationContainer>
        <FooterEnd>
            <P>
                <span>Made with </span>
                <FontAwesomeIcon icon={faHeart} /> 
                <span> by Yoni Sisso, Â©2020</span>
            </P>
            <IconContainer>
                <LinkContainer href='https://www.linkedin.com/in/yonisisso/' target='_blank' rel="noopener noreferrer">
                    <FontAwesomeIcon icon={faLinkedin} size="3x"/>
                </LinkContainer>
                <LinkContainer href='https://github.com/yonis9' target='_blank' rel="noopener noreferrer">
                    <FontAwesomeIcon icon={faGithubSquare} size="3x"/>
                </LinkContainer>
            </IconContainer>
        </FooterEnd>
    </FooterContainer>
)


const mapStateToProps = createStructuredSelector({
    currentUser: currentUserSelector
})

const mapDispatchToProps = dispatch => ({
    signOutStart: () => dispatch(signOutStart())
})


export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Footer));