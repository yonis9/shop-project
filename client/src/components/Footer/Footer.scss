.footer {
    background: #343A40;
    height: 300px;
    margin-top: 100px;
    color: #dae6ea;
    padding-top: 20px;
    font-family: 'Open Sans';


    .navigation-footer {
        height: 230px;
        display: flex;
        justify-content: space-around;
    }

    .left-navigation, .right-navigation {
        list-style: none;
        width: 200px;
    

        li {
            cursor: pointer;
            padding: 5px;

            &:hover {
                color: white;
            }
        }
    }


    .footer-end {
        
        padding: 0 50px;
        height: 70px;
        background-color: #212529;
        display: flex;
        align-items: center;
        justify-content: space-between;
        color: #868E96;

        p {
            font-size: 11px;
            
        }

        .icons-container {
            display: flex;
            justify-content: space-between;

            a {
                padding: 0 10px;
                transition: 0.2s ease-in;
                text-decoration: none;
                color: #868E96;

                &:hover {
                    cursor: pointer;
                    color: white;
                    transition: 0.2s ease-in;
                }
            }
        }
    }
}