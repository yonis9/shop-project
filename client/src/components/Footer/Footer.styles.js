import styled from 'styled-components';

export const FooterContainer = styled.div`
    background: #343A40;
    height: 300px;
    margin-top: 100px;
    color: #dae6ea;
    padding-top: 20px;
    font-family: 'Open Sans';
`;

export const NavigationContainer = styled.div`
    height: 230px;
    display: flex;
    justify-content: space-between;
    padding: 0 150px;
    @media only screen and (max-width: 800px) {
        padding: 0 20px;
    }

`;

export const LeftNavigation = styled.ul`
    list-style: none;
    
`;

export const RightNavigation = styled.ul`
    list-style: none;
    
`;


export const Navlink = styled.li`
    cursor: pointer;
    padding: 5px;

     &:hover {
        color: white;
        }
`;

Navlink.displayName = 'Navlink';


export const FooterEnd = styled.div`
    padding: 0 50px;
    height: 70px;
    background-color: #212529;
    display: flex;
    align-items: center;
    justify-content: space-between;
    color: #868E96;

    @media only screen and (max-width: 800px) {
        padding: 0 10px;
    }
`;

export const P = styled.p`
    font-size: 11px;
`;

export const IconContainer = styled.div`
    display: flex;
    justify-content: space-between;
`;

export const LinkContainer = styled.a`
    padding: 0 10px;
    transition: 0.2s ease-in;
    text-decoration: none;
    color: #868E96;

    &:hover {
        cursor: pointer;
        color: white;
        transition: 0.2s ease-in;
    }
`;