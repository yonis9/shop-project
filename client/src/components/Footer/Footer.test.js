import React from 'react';
import { shallow } from 'enzyme';

import { Footer } from './Footer.js';


describe('Footer component', () => {
    let wrapper;
    let mockHistory;
    let mockCurrentUser;
    let mockSignOutStart;

    beforeEach(() => {
        mockHistory = {
            push: jest.fn()
        };

        mockCurrentUser = {
            id: 1,
            name: 'Yoni'
        }

        mockSignOutStart = jest.fn();

       const mockProps = {
            history: mockHistory,
            currentUser: mockCurrentUser,
            signOutStart: mockSignOutStart
        };

        wrapper = shallow(<Footer {...mockProps} />);
    });

    it('should renderd Footer component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call history.push when clicked on LeftNavigation Navlinks', () => {
        for (let i = 0; i < 5; i++) {
            wrapper.find('Navlink').at(i).simulate('click');
        }
        
        expect(mockHistory.push).toHaveBeenCalledTimes(5);
    });

    it('should call signOutStart', () => {
        wrapper.find('Navlink').at(5).simulate('click');
        expect(mockSignOutStart).toHaveBeenCalled();
    });

    it('should call history.push when user is sign out', () => {
        const mockProps2 = { 
            history: mockHistory,
            signOutStart: mockSignOutStart,
            currentUser: null 
        };

        wrapper = shallow(<Footer {...mockProps2} />);

        wrapper.find('Navlink').at(5).simulate('click');
        expect(mockHistory.push).toHaveBeenCalled();
    })
});