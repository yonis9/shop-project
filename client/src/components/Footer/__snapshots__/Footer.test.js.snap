// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should renderd Footer component 1`] = `ShallowWrapper {}`;
